muikku-core/src/main/java/fi/otavanopisto/muikku/users/WorkspaceUserEntityController.java

```diff
@@ -205,6 +213,44 @@ public WorkspaceUserEntity updateIdentifier(WorkspaceUserEntity workspaceUserEnt
   }
 
   public WorkspaceUserEntity updateActive(WorkspaceUserEntity workspaceUserEntity, Boolean active) {
+    UserEntity userEntity = userEntityController.findUserEntityByDataSourceAndIdentifier(workspaceUserEntity.getUserSchoolDataIdentifier().getDataSource(), workspaceUserEntity.getUserSchoolDataIdentifier().getIdentifier());
+    UserEntityProperty lastWorkspaces = userEntityController.getUserEntityPropertyByKey(userEntity, "last-workspaces");
+    ObjectMapper objectMapper = new ObjectMapper();
+    List<LastWorkspace> lastWorkspaceList = null;
+    if (lastWorkspaces != null && active == Boolean.FALSE) {
+      if (lastWorkspaces.getValue() != null) {
+        List<LastWorkspace> newList = new ArrayList<LastWorkspace>();
+        String lastWorkspaceJson = lastWorkspaces.getValue();
+        
+        try {
+         lastWorkspaceList = objectMapper.readValue(lastWorkspaceJson, new TypeReference<ArrayList<LastWorkspace>>() {});
+        }
+        catch (JsonProcessingException e) {
+          // TODO Auto-generated catch block
+          e.printStackTrace();
+        }
+        
+        if(lastWorkspaceList != null) {
+          for (LastWorkspace lastWorkspace : lastWorkspaceList) {
+            if (lastWorkspace.getWorkspaceId().equals(workspaceUserEntity.getWorkspaceEntity().getId())) {
+              continue;
+            } else {
+              newList.add(lastWorkspace);
+            }
+          }
+        }
+        try {
+          if (newList.isEmpty()) {
```

Comment:
Shouldn't be needed. Front-end can most likely cope with `last-workspace` either being `null` or an empty array `[]`